cdk/apps/dev_worker/README.md

```diff
@@ -30,34 +30,101 @@ Since we run through the Fn.Sub to substitute variables into the user_data scrip
+|----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------|
+| EC2_TYPE             | Type of instance to be deployed, options can be seen here https://aws.amazon.com/ec2/instance-types/                                                                           | m4.4xlarge              |
+| MACHINE_IMAGE        | AMI can be found here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/finding-an-ami.html#finding-an-ami-console                                                          | ami-0dc96254d5535925f   |
+| VAR_VOLUME_SIZE      | /var is mounted on a separate EBS volume to stop 8 Gb root volume from filling up. Can also be used as a scratch space by setting TMPDIR to /var/tmp. Unit is Gb               | 16                      |
+| EXTENDED_VOLUME_SIZE | /data/ is also mounted on a separate EBS volume/. Unit is Gb                                                                                                                   | 100                     |
+| LAUNCH_TEMPLATE_NAME | If running a spot instance, this is the name of the launch template used                                                                                                       | dev_worker_template     |
+| INSTANCE_NAME        | This is the name of the instance deployed and will show up in the EC2 console                                                                                                  | dev_worker_instance_cdk |
+| USE_SPOT_INSTANCE    | Would you like to use a spot instance, much cheaper but cannot be stopped, only terminated and may be shutdown at any time or fail to launch if the MAX_SPOT_PRICE is too low. | True                    |
+| MAX_SPOT_PRICE       | Maximum hourly rate of the spot instance in dollar value                                                                                                                       | 0.30                    |
+
+## Validate the stack
 ```bash
-cdk deploy
+cdk synth -c "STACK_NAME=alexis-unique-stack" -c "KEY_NAME=myname-dev"
 ```
 
 ## Deploy the workflow with different parameters
 You can change any of the parameters seen in the `cdk.json` *context* attribute
 ```bash
-cdk diff -c "STACK_NAME=alexis-unique-stack" -c "KEY_NAME=myname-dev' -c "EC2_TYPE=t2.micro"
-cdk deploy -c "STACK_NAME=alexis-unique-stack" "KEY_NAME=myname-dev' -c "EC2_TYPE=t2.micro"
+cdk diff -c "STACK_NAME=alexis-unique-stack" -c "KEY_NAME=myname-dev" -c "EC2_TYPE=t2.micro"
+cdk deploy -c "STACK_NAME=alexis-unique-stack" -c "KEY_NAME=myname-dev" -c "EC2_TYPE=t2.micro"
 ```
 
 Notes:
-If you create a stack with a different *STACK_NAME* parameter, you will need to use this same context parameter when destorying the stack
+You will need to use this same STACK_NAME parameter when destroying the stack
```

Comment:
Is that really needed? 
I thought CDK creates a `cdk.context.json` with the current context. I would have hoped that a `cdk destroy` would take the context from that file without requiring the parameter again.

Reply:
I think you're right, but when I wrote this there was a default value for the STACK_NAME in `cdk.json` which was overwriting this hence it needed to be explicitly defined. I will re-test and see if the `try_get_context` method will also pick up from the `cdk.context.json`.

Reply:
If I run `cdk destroy` without the context parameter STACK_NAME, the `try_get_context` fails to retrieve it. Furthermore, from my directory, the `cdk.context.json` is not updated after running `cdk deploy -c "STACK_NAME=new_instance_name"

Reply:
Is the 'cdk.context.json` removed with `cdk destroy`?
Trying to figure out if perhaps the context is only generated on the first deploy and kept static...

Reply:
The `cdk.context.json` file in my directory hasn't been updated since March 3? It contains only the network information such as VPC ID, subnet ID and availability zone

Reply:
If you `cdk destroy`, then remove the `cdk.context.json` and `cdk deploy` again, does it show the same behaviour or does it generate the correct/updated context file?